compiler/describe/analyze.go

```diff
@@ -0,0 +1,145 @@
+package describe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/brimdata/zed/compiler/ast/dag"
+	"github.com/brimdata/zed/compiler/data"
+	"github.com/brimdata/zed/compiler/optimizer"
+	"github.com/brimdata/zed/lake"
+	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/pkg/field"
+	"github.com/segmentio/ksuid"
+)
+
+type Info struct {
+	Sources      []Source        `json:"sources"`
+	Aggregations []*Aggregation  `json:"aggregations"`
```

Comment:
Nit: Simplify this (unless you expect to add more fields to the Aggregation type down the road).
```suggestion
	AggregationKeys field.List   `json:"aggregation_keys"
```